src/mpid/common/shm/mpidu_init_shm.c

```diff
@@ -288,11 +290,42 @@ int MPIDU_Init_shm_query(int local_rank, void **target_addr)
     MPIR_Assert(MPIDU_Init_shm_local_size > 1);
 
     MPIR_Assert(local_rank < MPIDU_Init_shm_local_size);
-    *target_addr = (char *) baseaddr + local_rank * sizeof(MPIDU_Init_shm_block_t);
+    *target_addr = (char *) baseaddr + local_rank * MPIDU_INIT_SHM_BLOCK_SIZE;
 
     MPIR_FUNC_EXIT;
 
     return mpi_errno;
 }
 
+int MPIDU_Init_shm_atomic_count(void)
+{
+    return MPL_atomic_load_int(&barrier->alloc_count);
+}
```

Comment:
I'm not grokking the naming for this function. This is returning a boolean value of whether or not an allocation exists? Maybe `MPIDU_Init_shm_allocated` or `MPIDI_Init_shm_initialized`? Or could the value of alloc_count be more than 1 in the future?